pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -348,6 +348,10 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 	}
 
 	pipelineMeta, pipelineSpec, err := rprp.GetPipelineData(ctx, pr, getPipelineFunc)
+	if pipelineSpec != nil {
```

Comment:
There should be a test case testing this and should be corresponds to this change?

Reply:
yes I haven't finished all tests yet! Once done I will unhold this pr